source/extensions/omni.isaac.lab/omni/isaac/lab/scene/interactive_scene.py

```diff
@@ -365,6 +406,45 @@ def reset(self, env_ids: Sequence[int] | None = None):
         for sensor in self._sensors.values():
             sensor.reset(env_ids)
 
+    def reset_to(
+        self,
+        state: dict[str, dict[str, dict[str, torch.Tensor]]],
+        env_ids: Sequence[int] | None = None,
+        is_relative: bool = False,
+    ):
+        """Resets the scene entities to the given state.
+
+        Args:
+            state: The state to reset the scene entities to.
+            env_ids: The indices of the environments to reset.
+                Defaults to None (all instances).
+            is_relative: If set to True, the state is considered relative to the environment origins.
+        """
+        if env_ids is None:
+            env_ids = slice(None)
+        # -- assets
+        for asset_name, articulation in self._articulations.items():
+            joint_position = state["articulation"][asset_name]["joint_position"]
+            joint_velocity = state["articulation"][asset_name]["joint_velocity"]
+            articulation.write_joint_state_to_sim(joint_position, joint_velocity, env_ids=env_ids)
+            articulation.set_joint_position_target(joint_position, env_ids=env_ids)
+            articulation.set_joint_velocity_target(joint_velocity, env_ids=env_ids)
```

Comment:
similar to above, shouldn't we also set the root transform for non-fixed base robots?

Reply:
Added.